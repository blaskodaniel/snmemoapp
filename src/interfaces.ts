export interface NewMemo {
  DisplayName: string
  Description: string
}

export interface AddNewprops {
  show: boolean
  onCreate: (memo: NewMemo) => void
  onClose: () => void
}
