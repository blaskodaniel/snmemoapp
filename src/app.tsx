import React from 'react'
import { Container, CssBaseline, Grid } from '@material-ui/core'
import Toolbar from '@material-ui/core/Toolbar'
import snLogo from './assets/sensenet_logo_transparent.png'
import { NavBarComponent } from './components/navbar'
import { MemoPanel } from './components/memo-panel'

/**
 * The main entry point of your app. You can start h@cking from here ;)
 */
export const App: React.FunctionComponent = () => {
  return (
    <React.Fragment>
      <CssBaseline />
      <NavBarComponent />
      <Toolbar />
      <Container
        maxWidth="lg"
        style={{
          width: '100%',
          minHeight: '90vh',
          marginTop: '10px',
          display: 'flex',
          verticalAlign: 'middle',
          alignItems: 'center',
          justifyContent: 'center',
          flexDirection: 'column',
          backgroundPosition: 'center',
          backgroundRepeat: 'no-repeat',
          backgroundImage: `url(${snLogo})`,
          backgroundSize: 'auto',
        }}>
        <Grid container>
          <Grid item xs={12}>
            <MemoPanel />
          </Grid>
        </Grid>
      </Container>
    </React.Fragment>
  )
}
