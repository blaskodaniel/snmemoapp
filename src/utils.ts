export const tuple: any = <T extends string[]>(...args: T) => args
export const widgetTypes: string[] = tuple('markdown', 'query')
