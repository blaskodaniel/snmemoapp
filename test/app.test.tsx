import { shallow } from 'enzyme'
import React from 'react'
import { App } from '../src/app'

describe('The app layout instance', () => {
  it('should renders correctly', () => {
    expect(shallow(<App />)).toMatchSnapshot()
  })
})
