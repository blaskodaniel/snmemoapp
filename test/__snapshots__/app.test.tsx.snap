// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The app layout instance should renders correctly 1`] = `
<Fragment>
  <CssBaseline />
  <Component />
  <WithStyles(ForwardRef(Toolbar)) />
  <WithStyles(ForwardRef(Container))
    maxWidth="lg"
    style={
      Object {
        "alignItems": "center",
        "backgroundImage": "url(sensenet_logo_transparent.png)",
        "backgroundPosition": "center",
        "backgroundRepeat": "no-repeat",
        "backgroundSize": "auto",
        "display": "flex",
        "flexDirection": "column",
        "justifyContent": "center",
        "marginTop": "10px",
        "minHeight": "90vh",
        "verticalAlign": "middle",
        "width": "100%",
      }
    }
  >
    <WithStyles(ForwardRef(Grid))
      container={true}
    >
      <WithStyles(ForwardRef(Grid))
        item={true}
        xs={12}
      >
        <Component />
      </WithStyles(ForwardRef(Grid))>
    </WithStyles(ForwardRef(Grid))>
  </WithStyles(ForwardRef(Container))>
</Fragment>
`;
